src/output-json-tls.c

```diff
@@ -365,6 +376,88 @@ static void JsonTlsLogClientCert(
     }
 }
 
+static void JsonTlsLogClientHandshake(SCJsonBuilder *js, SSLState *ssl_state)
+{
+    const uint16_t *val;
+    uintptr_t i, nr;
+
+    if (ssl_state->client_connp.hs == NULL) {
+        return;
+    }
+
+    // Don't write an empty handhsake
+    if (SCHandshakeIsEmpty(ssl_state->client_connp.hs)) {
+        return;
+    }
+
+    SCJbOpenObject(js, "client_handshake");
+
+    const uint16_t vers = SCHandshakeGetVersion(ssl_state->client_connp.hs);
+    JsonTlsLogVersion(js, vers);
+
+    val = SCHandshakeGetCiphers(ssl_state->client_connp.hs, &nr);
```

Comment:
could this move into rust? It seems we have quite a bit of back and forth between C and Rust here, while everything we need is easily accessible directly in rust?

Reply:
Are you meaning for the rust side to write the object output as json? Could you clarify your meaning further? I think this gives us a bit more freedom because the `HandshakeParams` object is used to hold information for client and server. It's then the responsibility of the C side to display whichever components it sees fit. The handshake object is 'generic' in that sense, we wouldn't want the json output, for example, to be generic. With the current design it's more specialised for each 'component' where component = client/server handshake. But maybe I've misunderstood your intention?

Reply:
My idea is to replace these blocks by a simple call like
```
SCTLSHandshakeLogCiphers(js, ssl_state->client_connp.hs);
```
In the current code we're in C, working on a JsonBuilder object (rust), iterating a Handshake cipher list (rust), building a jb array (rust). So to me it seems like the logical next step would be to move that to rust.

Reply:
perhaps we can move more of this func to rust btw, it was the array building that triggered my attention

Reply:
Great! Thanks, having a look now..